vmss.tf

```diff
@@ -161,11 +161,12 @@ resource "azurerm_role_assignment" "vm_role_assignment" {
 resource "null_resource" "force-delete-vmss" {
   triggers = {
     vmss_name       = azurerm_linux_virtual_machine_scale_set.vmss.name
-    rg_name         = var.rg_name
+    rg_name         = data.azurerm_resource_group.rg.name
```

Comment:
why did you need to change this?

Reply:
what makes it better?

Reply:
from data you sure u get get the right rg

Reply:
Isn't it the same?
You use here the var:
```
data azurerm_resource_group "rg"{
  name = var.rg_name
}
```
Also it requires additional api call.